src/components/lists/items/ActionListItem.jsx

```diff
@@ -103,16 +103,26 @@ export default class ActionListItem extends React.Component {
 
     componentDidUpdate(prevProps) {
         if (this.props.inView && !prevProps.inView) {
-            const action = this.props.data;
-            const participants = this.props.participants;
-            const responses = this.props.responses;
+            this.loadSubData();
+        }
+    }
 
-            if (!participants) {
-                this.props.dispatch(retrieveActionParticipants(action.id));
-            }
-            if (!responses) {
-                this.props.dispatch(retrieveActionResponses(action.id));
-            }
+    componentDidMount() {
+        if (this.props.inView) {
+            this.loadSubData();
+        }
+    }
+
+    loadSubData() {
+        const action = this.props.data;
+        const participants = this.props.participants;
+        const responses = this.props.responses;
```

Comment:
Is destructuring assignment available here?

`const { action, participants, responses } = this.props;`

Reply:
It should work, but it's something we haven't been using (consistently) before. I really think we need a general code style overhaul at some point seeing how javascript has evolved since this project started (almost exactly!) four years ago. That's the kind of work that I have planned for late 2019, although I guess it makes sense to start implementing it little by little already.

Reply:
(Thanks also for spotting the bug in my shoddy example!)